### PR TITLE
Update component lifecycle doc

### PR DESCRIPTION
**Pull request by broccolinisoup**
Closes https://github.com/github/primer/issues/1520

Updates the testing criteria.

### CODE REVIEW

File: content/component-lifecycle.md [hunk 1]
```diff
@@ -17,9 +17,10 @@ The component is ready for preliminary usage, with breaking changes expected.
...
 - The component is compatible with color modes and can adapt to different themes. The component does not reference any hard-coded static values and uses functional variables. Any temporary or app-level variables have been removed. Theme values (such as a new brand scheme) can be swapped out without touching the public API of the component.
 - The component is designed with responsiveness in mind. The component can adapt to different screen sizes, from mobile to large desktop screens. The component's interactive areas must be friendly to touch on devices with a coarse pointer method.
-- Basic documentation exists that includes example usage of the component.
+- Basic documentation that includes example usage of the component exists on [primer.style](https://primer.style/).
 - Primary use cases tested and reviewed.
-- The component has 100% test coverage.
+- The component has 100% unit test coverage (where applicable), visual regression tests and interaction tests.
+- The component does not introduce any [axe](https://www.deque.com/axe/) violations.
```
Comment (broccolinisoup) [inline content/component-lifecycle.md:23]: There is no criteria about accessibility for Alpha. I took it into its own bullet - let me know if that doesn't makes sense 🙌🏻

File: content/component-lifecycle.md [hunk 2]
```diff
@@ -17,9 +17,10 @@ The component is ready for preliminary usage, with breaking changes expected.
 - The component does not have external dependencies. This could be external libraries or dependencies on libraries in an application (like GitHub.com)
 - The component is compatible with color modes and can adapt to different themes. The component does not reference any hard-coded static values and uses functional variables. Any temporary or app-level variables have been removed. Theme values (such as a new brand scheme) can be swapped out without touching the public API of the component.
 - The component is designed with responsiveness in mind. The component can adapt to different screen sizes, from mobile to large desktop screens. The component's interactive areas must be friendly to touch on devices with a coarse pointer method.
-- Basic documentation exists that includes example usage of the component.
+- Basic documentation that includes example usage of the component exists on [primer.style](https://primer.style/).
 - Primary use cases tested and reviewed.
-- The component has 100% test coverage.
+- The component has 100% unit test coverage (where applicable), visual regression tests and interaction tests.
```
Comment (rezrah) [inline content/component-lifecycle.md:22]: This is great! Can we use this opportunity to elaborate on these statements?

Also wonder if we can change the term "interaction tests", as that terminology relates to the Storybook-specific paradigms. Can we use a more generic STLC term like "end-to-end" or "open-box testing"?

```suggestion
- The component has robust unit test coverage on all critical paths, including branching logic and edge cases. This is measurable through metrics like 100% test coverage (where achievable). 
- The component has adequate visual regression coverage for its default and critical interactive states.
- The component has robust test coverage for its interactive (stateful) behaviour, which has been verified in end-to-end and open-box testing environments.
```

File: content/component-lifecycle.md [hunk 3]
```diff
@@ -17,9 +17,10 @@ The component is ready for preliminary usage, with breaking changes expected.
 - The component does not have external dependencies. This could be external libraries or dependencies on libraries in an application (like GitHub.com)
 - The component is compatible with color modes and can adapt to different themes. The component does not reference any hard-coded static values and uses functional variables. Any temporary or app-level variables have been removed. Theme values (such as a new brand scheme) can be swapped out without touching the public API of the component.
 - The component is designed with responsiveness in mind. The component can adapt to different screen sizes, from mobile to large desktop screens. The component's interactive areas must be friendly to touch on devices with a coarse pointer method.
-- Basic documentation exists that includes example usage of the component.
+- Basic documentation that includes example usage of the component exists on [primer.style](https://primer.style/).
```
Comment (joelhawksley) [inline content/component-lifecycle.md:20]: In the new docs IA, PVC examples might not be on Primer.style. I'd suggest:

```suggestion
- Component documentation includes example usage of the component.
```

Comment (broccolinisoup) [reply content/component-lifecycle.md:20]: Ah great point! Thanks 🙌🏻

File: content/component-lifecycle.md (hunk 2 above)
Comment (broccolinisoup) [reply content/component-lifecycle.md:22]: I love it 😍 Beautifully written! Thanks @rezrah 🙌🏻

**Review by lesliecdubs — APPROVED** [review]
Good feedback from the other reviewers. Once resolved, this LGTM.

**Review by rezrah — APPROVED** [review]
🚀
